File: dart/resistor-color/test/resistor_color_test.dart
import 'package:resistor_color/resistor_color.dart';
import 'package:test/test.dart';

final resistorColor = ResistorColor();

void main() {
  group('ResistorColor', () {
    group('Color codes', () {
      test('Black', () {
        final int result = resistorColor.colorCode('black');
        expect(result, equals(0));
      }, skip: false);

      test('White', () {
        final int result = resistorColor.colorCode('white');
        expect(result, equals(9));
      });

      test('Orange', () {
        final int result = resistorColor.colorCode('orange');
        expect(result, equals(3));
      });
    });

    test('Colors', () {
      final List<String> result = resistorColor.colors;
      expect(
          result,
          equals(<String>[
            'black',
            'brown',
            'red',
            'orange',
            'yellow',
            'green',
            'blue',
            'violet',
            'grey',
            'white',
          ]));
    });
  });
}

File: ruby/resistor-color/resistor_color_test.rb
require 'minitest/autorun'
require_relative 'resistor_color'

# Common test data version: 1.0.0 edf1778
class ResistorColorTest < Minitest::Test
  def test_black
    # 
    assert_equal 0, ResistorColor.color_code("black")
  end

  def test_white
    
    assert_equal 9, ResistorColor.color_code("white")
  end

  def test_orange
    
    assert_equal 3, ResistorColor.color_code("orange")
  end

  def test_colors
    
    expected = %w[black brown red orange yellow green blue violet grey white]
    assert_equal expected, ResistorColor::COLORS
  end
end
